pkg/devfile/resources.go

```diff
@@ -0,0 +1,237 @@
+package devfile
+
+import (
+	"fmt"
+	"strconv"
+
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/library/pkg/devfile/generator"
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/devfile/library/pkg/devfile/parser/data/v2/common"
+	"github.com/hashicorp/go-multierror"
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/spf13/afero"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog"
+)
+
+// GetBuildResource gets the build config resource
+func GetBuildResource(data DevfileForm, dockerfilePath, contextDir string) buildv1.BuildConfig {
+
+	buildConfigParams := generator.BuildConfigParams{
+		TypeMeta: generator.GetTypeMeta("BuildConfig", "build.openshift.io/v1"),
+		BuildConfigSpecParams: generator.BuildConfigSpecParams{
+			ImageStreamTagName: data.Name + ":latest", // TODO Post Dev Preview. Update as per proposal i.e.; use the image mentioned in the devfile and push build to it.
+			GitRef:             data.Git.Ref,
+			GitURL:             data.Git.URL,
+			ContextDir:         contextDir,
+			BuildStrategy:      generator.GetDockerBuildStrategy(dockerfilePath, nil), // TODO use the Dockerfile path from the devfile instead of assuming
+		},
+	}
+
+	buildConfig := generator.GetBuildConfig(buildConfigParams)
+
+	return *buildConfig
+}
+
+// GetDeployResource gets the deployment resource
+func GetDeployResource(name string, devfileObj parser.DevfileObj, filterOptions common.DevfileOptions) (appsv1.Deployment, error) {
```

Comment:
Why we can't use what is already in there in `devfile/library`?  (https://github.com/devfile/library/blob/a7dbafc0c2807db47d9ee7fa2a1cf19998b0236f/pkg/devfile/generator/generators.go#L180)